src/parsing/text_reader.rs

```diff
@@ -28,82 +30,81 @@ where
         TextReader {
             inner,
             decoder: charset.new_decoder(),
+            internal_buf: vec![0u8; 4096],
+            internal_buf_amt: 0,
             eof: false,
         }
     }
 }
 
-impl<R> fmt::Debug for TextReader<R>
-where
-    R: fmt::Debug + BufRead,
-{
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.debug_struct("TextReader")
-            .field("inner", &self.inner)
-            .field("decoder", &"<Decoder>")
-            .field("eof", &self.eof)
-            .finish()
-    }
-}
+// impl<R> fmt::Debug for TextReader<R>
+// where
+//     R: fmt::Debug + BufRead,
+// {
+//     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+//         f.debug_struct("TextReader")
+//             .field("inner", &self.inner)
+//             .field("decoder", &"<Decoder>")
+//             .field("eof", &self.eof)
+//             .finish()
+//     }
+// }
 
 impl<R> Read for TextReader<R>
 where
     R: BufRead,
 {
-    fn read(&mut self, mut buf: &mut [u8]) -> io::Result<usize> {
-        if self.eof {
+    fn read(&mut self, mut dst: &mut [u8]) -> io::Result<usize> {
+        // The `Read` trait is not directly compatible with the `encoding_rs` crate. There is no
```

Comment:
But is that really our problem? `Read::read` never guarantees to write the full buffer anyway and passing in a buffer so small that it cannot contain a single code point would then yield the unhelpful but correct result of reading zero bytes.

If we do go for the internal buffer here, then I'd say that it make little sense to implement `io::Read` and we should go for `io::BufRead` instead (with `io::Read` using the trivial implementation as `ChunkedReader` does).

Reply:
We could also consider just replacing our dependency on `encoding_rs` by [`encoding_rs_io`](https://docs.rs/encoding_rs_io) or if we want to stick to `Read` at least use the same trick as they do with their [`TinyTranscoder`](https://docs.rs/crate/encoding_rs_io/0.1.6/source/src/util.rs), i.e. use a small fixed-size overflow buffer.

Reply:
Created #34 to show case the `encoding_rs_io` route.